x/admin/keeper/msg_server.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sagaxyz/saga-sdk/x/admin/types"
+)
+
+var _ types.MsgServer = &Keeper{}
+
+var ErrNotAuthorized = errors.New("not authorized")
+var ErrInvalidRequest = errors.New("invalid request")
+
+func (k Keeper) SetMetadata(
+	goCtx context.Context,
+	msg *types.MsgSetMetadata,
+) (*types.MsgSetMetadataResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if msg.Metadata == nil {
+		return nil, errorsmod.Wrap(ErrInvalidRequest, "metadata is nil")
+	}
+
```

Comment:
if there is some metadata provided, do we need to validate it here? or is that done in the `ValidateBasic()`?